sway/criteria.c

```diff
@@ -366,50 +371,80 @@ static enum criteria_token token_from_name(char *name) {
  * using criteria via IPC. Using __focused__ in config is not useful because
  * criteria is only executed once per view.
  */
-static char *get_focused_prop(enum criteria_token token) {
+static char *get_focused_prop(enum criteria_token token, bool *autofail) {
 	struct sway_seat *seat = input_manager_current_seat();
 	struct sway_container *focus = seat_get_focused_container(seat);
 
-	if (!focus || !focus->view) {
-		return NULL;
-	}
-	struct sway_view *view = focus->view;
+	struct sway_view *view = focus ? focus->view : NULL;
 	const char *value = NULL;
 
 	switch (token) {
 	case T_APP_ID:
-		value = view_get_app_id(view);
+		if (view) {
+			value = view_get_app_id(view);
+		}
+		if (!value) {
+			*autofail = true;
```

Comment:
We don't need those `if (!value)` anymore

Reply:
Oh right. `parse_token` checks both the returned value and `autofail` so setting `autofail` even when there is a value is fine.